src/main/java/service/shop/ShopServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package service.shop;
+
+import java.util.List;
+import model.FruitTransaction;
+import service.operation.OperationHandler;
+import service.operation.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler operationHandler = operationStrategy.get(transaction.getOperation());
+            operationHandler.operation(transaction.getOperation(), transaction);
+        }
```

Comment:
While the use of the Strategy pattern here is good, consider improving the encapsulation of the `OperationHandler` instances. Instead of retrieving a new handler for each transaction, you could initialize all required handlers once and reuse them, which would be more efficient and cleaner. This would also ensure that your service is more independent and focused on a single problem, adhering to the SOLID principles.